# Question : As we have learned to find out statistical test like mean median standard deviation using python so practice those methods and using the results write down 10 hypothesis and submit.

# Analysis for Hypothesis

In [14]:
import pandas as pd
df = pd.read_csv('E:\Riyaz\LetsUpgrade-AI-ML\Code\general_data.csv').dropna().drop_duplicates()
df.head()

Age Attrition     BusinessTravel              Department  DistanceFromHome  \
0   51        No      Travel_Rarely                   Sales                 6   
1   31       Yes  Travel_Frequently  Research & Development                10   
2   32        No  Travel_Frequently  Research & Development                17   
3   38        No         Non-Travel  Research & Development                 2   
4   32        No      Travel_Rarely  Research & Development                10   

   Education EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0          2  Life Sciences              1           1  Female  ...   
1          1  Life Sciences              1           2  Female  ...   
2          4          Other              1           3    Male  ...   
3          5  Life Sciences              1           4    Male  ...   
4          1        Medical              1           5    Male  ...   

   NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0                 1.0      Y                11              8   
1                 0.0      Y                23              8   
2                 1.0      Y                15              8   
3                 3.0      Y                11              8   
4                 4.0      Y                12              8   

   StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  YearsAtCompany  \
0                 0               1.0                      6               1   
1                 1               6.0                      3               5   
2                 3               5.0                      2               5   
3                 3              13.0                      5               8   
4                 2               9.0                      2               6   

   YearsSinceLastPromotion  YearsWithCurrManager  
0                        0                     0  
1                        1                     4  
2                        0                     3  
3                        7                     5  
4                        0                     4  

[5 rows x 24 columns]

In [15]:
df.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

In [16]:
df.isnull()

Age  Attrition  BusinessTravel  Department  DistanceFromHome  \
0     False      False           False       False             False   
1     False      False           False       False             False   
2     False      False           False       False             False   
3     False      False           False       False             False   
4     False      False           False       False             False   
...     ...        ...             ...         ...               ...   
4404  False      False           False       False             False   
4405  False      False           False       False             False   
4406  False      False           False       False             False   
4407  False      False           False       False             False   
4408  False      False           False       False             False   

      Education  EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0         False           False          False       False   False  ...   
1         False           False          False       False   False  ...   
2         False           False          False       False   False  ...   
3         False           False          False       False   False  ...   
4         False           False          False       False   False  ...   
...         ...             ...            ...         ...     ...  ...   
4404      False           False          False       False   False  ...   
4405      False           False          False       False   False  ...   
4406      False           False          False       False   False  ...   
4407      False           False          False       False   False  ...   
4408      False           False          False       False   False  ...   

      NumCompaniesWorked  Over18  PercentSalaryHike  StandardHours  \
0                  False   False              False          False   
1                  False   False              False          False   
2                  False   False              False          False   
3                  False   False              False          False   
4                  False   False              False          False   
...                  ...     ...                ...            ...   
4404               False   False              False          False   
4405               False   False              False          False   
4406               False   False              False          False   
4407               False   False              False          False   
4408               False   False              False          False   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
0                False              False                  False   
1                False              False                  False   
2                False              False                  False   
3                False              False                  False   
4                False              False                  False   
...                ...                ...                    ...   
4404             False              False                  False   
4405             False              False                  False   
4406             False              False                  False   
4407             False              False                  False   
4408             False              False                  False   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0              False                    False                 False  
1              False                    False                 False  
2              False                    False                 False  
3              False                    False                 False  
4              False                    False                 False  
...              ...                      ...                   ...  
4404           False                    False                 False  
4405           False    

In [17]:
df.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
4404    False
4405    False
4406    False
4407    False
4408    False
Length: 4382, dtype: bool

In [18]:
df.drop_duplicates()

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4404   29        No      Travel_Rarely                   Sales   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4404                 4          3          Other              1        4405   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4404  Female  ...                 1.0      Y                18              8   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4404                 0               6.0                      2   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...                   ..

## Original data

In [19]:
df[['Age','DistanceFromHome','Education','MonthlyIncome',
'NumCompaniesWorked', 'PercentSalaryHike','TotalWorkingYears', 'TrainingTimesLastYear',
'YearsAtCompany','YearsSinceLastPromotion', 'YearsWithCurrManager']].describe()

Age  DistanceFromHome    Education  MonthlyIncome  \
count  4382.000000       4382.000000  4382.000000    4382.000000   
mean     36.933364          9.198996     2.912369   65061.702419   
std       9.137272          8.105396     1.024728   47142.310175   
min      18.000000          1.000000     1.000000   10090.000000   
25%      30.000000          2.000000     2.000000   29110.000000   
50%      36.000000          7.000000     3.000000   49190.000000   
75%      43.000000         14.000000     4.000000   83790.000000   
max      60.000000         29.000000     5.000000  199990.000000   

       NumCompaniesWorked  PercentSalaryHike  TotalWorkingYears  \
count         4382.000000        4382.000000        4382.000000   
mean             2.693291          15.210634          11.290278   
std              2.497832           3.663007           7.785717   
min              0.000000          11.000000           0.000000   
25%              1.000000          12.000000           6.000000   
50%              2.000000          14.000000          10.000000   
75%              4.000000          18.000000          15.000000   
max              9.000000          25.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count            4382.000000     4382.000000              4382.000000   
mean                2.798266        7.010497                 2.191693   
std                 1.289402        6.129351                 3.224994   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        3.000000                 0.000000   
50%                 3.000000        5.000000                 1.000000   
75%                 3.000000        9.000000                 3.000000   
max                 6.000000       40.000000                15.000000   

       YearsWithCurrManager  
count           4382.000000  
mean               4.126198  
std                3.569674  
min                0.000000  
25%                2.000000  
50%                3.000000  
75%                7.000000  
max               17.000000

## Attrited

In [20]:
df_1 = df.loc[df['Attrition'] == 'Yes']
df_1_desc=df_1[['Age','DistanceFromHome','Education','MonthlyIncome',
'NumCompaniesWorked', 'PercentSalaryHike','TotalWorkingYears', 'TrainingTimesLastYear',
'YearsAtCompany','YearsSinceLastPromotion', 'YearsWithCurrManager']].describe()
df_1_desc

Age  DistanceFromHome   Education  MonthlyIncome  \
count  705.000000        705.000000  705.000000     705.000000   
mean    33.628369          9.024113    2.872340   61814.950355   
std      9.678836          7.755184    1.014463   44890.457122   
min     18.000000          1.000000    1.000000   10090.000000   
25%     28.000000          2.000000    2.000000   28440.000000   
50%     32.000000          7.000000    3.000000   49080.000000   
75%     39.000000         15.000000    4.000000   71040.000000   
max     58.000000         29.000000    5.000000  198590.000000   

       NumCompaniesWorked  PercentSalaryHike  TotalWorkingYears  \
count          705.000000         705.000000         705.000000   
mean             2.937589          15.487943           8.273759   
std              2.681283           3.785842           7.176762   
min              0.000000          11.000000           0.000000   
25%              1.000000          12.000000           3.000000   
50%              1.000000          14.000000           7.000000   
75%              5.000000          18.000000          10.000000   
max              9.000000          25.000000          40.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count             705.000000      705.000000               705.000000   
mean                2.658156        5.148936                 1.960284   
std                 1.155899        5.960972                 3.157532   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        1.000000                 0.000000   
50%                 3.000000        3.000000                 1.000000   
75%                 3.000000        7.000000                 2.000000   
max                 6.000000       40.000000                15.000000   

       YearsWithCurrManager  
count            705.000000  
mean               2.865248  
std                3.146562  
min                0.000000  
25%                0.000000  
50%                2.000000  
75%                5.000000  
max               14.000000

## Not attrited

In [21]:
df_2 = df.loc[df['Attrition'] == 'No']
df_2[['Age','DistanceFromHome','Education','MonthlyIncome',
'NumCompaniesWorked', 'PercentSalaryHike','TotalWorkingYears', 'TrainingTimesLastYear',
'YearsAtCompany','YearsSinceLastPromotion', 'YearsWithCurrManager']].describe()

Age  DistanceFromHome    Education  MonthlyIncome  \
count  3677.000000       3677.000000  3677.000000    3677.000000   
mean     37.567038          9.232527     2.920044   65684.209954   
std       8.891816          8.171419     1.026644   47542.428949   
min      18.000000          1.000000     1.000000   10510.000000   
25%      31.000000          2.000000     2.000000   29360.000000   
50%      36.000000          7.000000     3.000000   49300.000000   
75%      43.000000         14.000000     4.000000   86060.000000   
max      60.000000         29.000000     5.000000  199990.000000   

       NumCompaniesWorked  PercentSalaryHike  TotalWorkingYears  \
count         3677.000000        3677.000000        3677.000000   
mean             2.646451          15.157465          11.868643   
std              2.458712           3.637095           7.765251   
min              0.000000          11.000000           0.000000   
25%              1.000000          12.000000           6.000000   
50%              2.000000          14.000000          10.000000   
75%              4.000000          18.000000          16.000000   
max              9.000000          25.000000          38.000000   

       TrainingTimesLastYear  YearsAtCompany  YearsSinceLastPromotion  \
count            3677.000000     3677.000000              3677.000000   
mean                2.825129        7.367419                 2.236062   
std                 1.311886        6.097294                 3.236300   
min                 0.000000        0.000000                 0.000000   
25%                 2.000000        3.000000                 0.000000   
50%                 3.000000        6.000000                 1.000000   
75%                 3.000000       10.000000                 3.000000   
max                 6.000000       37.000000                15.000000   

       YearsWithCurrManager  
count           3677.000000  
mean               4.367963  
std                3.595395  
min                0.000000  
25%                2.000000  
50%                3.000000  
75%                7.000000  
max               17.000000

## Number of Companies Worked more than 4

In [22]:
df_3 = df_1.loc[df_1['NumCompaniesWorked'] > 4]
df_3[['Age','DistanceFromHome','Education','MonthlyIncome',
 'PercentSalaryHike','TotalWorkingYears', 'TrainingTimesLastYear',
'YearsAtCompany','YearsSinceLastPromotion', 'YearsWithCurrManager']].describe()

Age  DistanceFromHome   Education  MonthlyIncome  \
count  199.000000        199.000000  199.000000     199.000000   
mean    36.994975          8.552764    3.085427   58937.135678   
std      9.713833          7.114928    0.946945   46344.372070   
min     22.000000          1.000000    1.000000   10090.000000   
25%     29.000000          2.000000    3.000000   27410.000000   
50%     34.000000          7.000000    3.000000   45220.000000   
75%     43.000000         11.500000    4.000000   67120.000000   
max     58.000000         28.000000    5.000000  196270.000000   

       PercentSalaryHike  TotalWorkingYears  TrainingTimesLastYear  \
count         199.000000         199.000000             199.000000   
mean           15.844221          10.849246               2.562814   
std             4.134220           7.233996               1.203876   
min            11.000000           3.000000               0.000000   
25%            12.000000           6.000000               2.000000   
50%            15.000000           8.000000               2.000000   
75%            19.000000          15.000000               3.000000   
max            25.000000          40.000000               6.000000   

       YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
count      199.000000               199.000000            199.000000  
mean         5.306533                 1.989950              2.924623  
std          5.874230                 3.231771              2.737570  
min          0.000000                 0.000000              0.000000  
25%          2.000000                 0.000000              0.500000  
50%          4.000000                 1.000000              2.000000  
75%          6.000000                 2.000000              4.000000  
max         31.000000                14.000000             10.000000

## Number of trainings taken last year more than 3

In [23]:
df_4 = df_1.loc[df_1['TrainingTimesLastYear'] > 3]
df_4[['Age','DistanceFromHome','Education','MonthlyIncome',
 'PercentSalaryHike','TotalWorkingYears',
'YearsAtCompany','YearsSinceLastPromotion', 'YearsWithCurrManager']].describe()

Age  DistanceFromHome   Education  MonthlyIncome  \
count  110.000000        110.000000  110.000000     110.000000   
mean    30.045455         12.136364    3.000000   58312.090909   
std      8.003883          8.295547    1.022679   46181.724732   
min     18.000000          1.000000    1.000000   17900.000000   
25%     26.000000          6.000000    3.000000   28530.000000   
50%     29.000000          9.000000    3.000000   36920.000000   
75%     33.000000         19.000000    4.000000   66520.000000   
max     53.000000         29.000000    4.000000  187890.000000   

       PercentSalaryHike  TotalWorkingYears  YearsAtCompany  \
count         110.000000         110.000000      110.000000   
mean           16.290909           6.263636        3.654545   
std             4.324764           4.528558        3.192412   
min            11.000000           0.000000        0.000000   
25%            12.000000           4.000000        2.000000   
50%            16.000000           6.000000        3.000000   
75%            19.000000           8.000000        5.000000   
max            25.000000          23.000000       11.000000   

       YearsSinceLastPromotion  YearsWithCurrManager  
count               110.000000            110.000000  
mean                  1.609091              2.518182  
std                   2.287162              2.801480  
min                   0.000000              0.000000  
25%                   0.000000              1.000000  
50%                   1.000000              2.000000  
75%                   2.000000              2.000000  
max                   9.000000             10.000000

# Hypothesis Statements

## Hypothesis 1

Null Hypothesis: 
- The mean age of an employee who has attrited is equal to 32

Alternative Hypothesis:
- The mean age of an employee who has attrited is not equal to 32

The age of an employee who has attrited has a standard deviation of 9.67.

To determine if this is true, a sample of 70 attrited employees was taken and their mean age was found to be 34.

## Hypothesis 2

Null Hypothesis: 
- The mean monthly income of attrited employees is equal to Rs.60,000 

Alternative Hypothesis:
- The mean monthly income of attrited employees is not equal to Rs.60,000.

To determine if this is true, a sample of 70 attrited employees was taken and their mean monthly income was found to be Rs.64,650.
 
The monthly income of attrited employees has a standard deviation of 44890



## Hypothesis 3


 Null Hypothesis:
- The mean years at company of attrited employees is greater than or equal to 3 years 

Alternative Hypothesis:
- The mean years at company of attrited employees is less than 3 years 

To determine if this is true,a sample of 70 attrited employees was taken at random and their mean years at the comapny was found to be 4.44 years

The years at company of attrited employees has a standard deviation of 5.96


## Hypothesis 4

Null Hypothesis:
- The mean age of employees who have not attrited is equal to 38 years 

Alternative Hypothesis:
- The mean age of employees who have not attrited is not equal to 38 years

In order to test this claim, a sample of 70 employees who have not attrited was taken at random and their mean age was found to be 36

The age of employees who have not attrited has a standard devaition of 8.89

## Hypothesis 5

Null Hypothesis:
- Mean of total working years of attrited employees is less than or equal to 7 years

Alternative Hypothesis:
- Mean of total working years of attrited employees is greater than 7 years

In order to test this claim, a sample of 70 employees who have attrited was taken at random and their mean working years was found to be 8

Total working years of attrited employees has a standard deviation of 7.17


## Hypothesis 6

Null Hypothesis:
- The average number of years spent by an attrited employee with current manager is equal to 4 

Alternative Hypothesis:
- The average number of years spent by an attrited employee with current manager is not equal to 4 

In order to test this claim, a sample of 25 employees who have attrited was taken at random and their average number of years spent with current manager was found to be equal to 2.12

The number of years spent by an attrited employee with current manager has a standard deviation of 3.14

## Hypothesis 7

Null Hypothesis:
- The average number of years spent by a non-attrited employee with current manager is greater than or equal to 4 

Alternative Hypothesis:
- The average number of years spent by a non-attrited employee with current manager is less than 4

In order to test this claim, a sample of 70 employees who have not attrited was taken at random and their average number of years spent with current manager was found to be equal to 3.80

The number of years spent by a non-attrited employee with current manager has a standard deviation of 3.59

## Hypothesis 8

Null Hypothesis:
- The mean pecentage in salary hike of the attrited employees is greater than or equal to 17% 

Alternative Hypothesis:
- The mean pecentage in salary hike of the attrited employees is less than 17

In order to test this claim, a sample of 25 employees who have attrited was taken at random and mean pecentage in their salary hike was found to be equal to 16.01%

The pecentage in salary hike of the attrited employees has a standard deviation of 3.78

## Hypothesis 9

Null Hypothesis:
- The mean salary of attrited employees who have worked in more than 4 companies is equal to Rs.56,000 

Alternative Hypothesis:
- The mean salary of attrited employees who have worked in more than 4 companies is not equal to Rs.56,000

In order to test this claim, a sample of 100 employees who have attrited and worked in more than 4 companies was taken at random and mean salary was found to be Rs 57,955

The salary of attrited employees who have worked in more than 4 companies has a standard deviation of 46344


## Hypothesis 10

Null Hypothesis:
- The mean pecentage in salary hike of the attrited employees who took training more than 3 times in the previous year is found to be greater than or equal to 18% 

Alternative Hypothesis:
- The mean pecentage in salary hike of the attrited employees who took training more than 3 times in the previous year is found to be less than 18%

In order to test this claim, a sample of 25 employees who have attrited and took training more than 3 times in the previous year was taken at random and their mean percentage in salary hike was found to be 17.28%

The pecentage in salary hike of the attrited employees who took training more than 3 times in the previous year has a standard deviation of 4.32